Error upload to AppStore TestFlight using fastlane v2.198.0 (An error occurred while processing the json request. (1268)) using Azure Devops

### Issue Description
Can not upload .ipa to TestFlight with version 2.198.0 but when I switch back to 2.197.0 it working...


##### Complete output when running fastlane, including the stack trace and command used


2021-11-16T12:58:46.6255080Z ==============================================================================
2021-11-16T12:58:46.6255470Z Task         : Apple App Store Release
2021-11-16T12:58:46.6255790Z Description  : Release an app to TestFlight or the Apple App Store
2021-11-16T12:58:46.6256080Z Version      : 1.189.0
2021-11-16T12:58:46.6256320Z Author       : Microsoft Corporation
2021-11-16T12:58:46.6257840Z Help         : [More Information](https://marketplace.visualstudio.com/items?itemName=ms-vsclient.app-store)
2021-11-16T12:58:46.6258310Z ==============================================================================
2021-11-16T12:58:49.1510240Z [command]/usr/local/opt/ruby@2.7/bin/gem install fastlane
2021-11-16T12:59:23.0450540Z Successfully installed fastlane-2.198.0
2021-11-16T12:59:23.0451520Z Parsing documentation for fastlane-2.198.0
2021-11-16T12:59:23.0457530Z Installing ri documentation for fastlane-2.198.0
2021-11-16T12:59:23.0458260Z Done installing documentation for fastlane after 29 seconds
2021-11-16T12:59:23.0458640Z 1 gem installed
2021-11-16T12:59:23.0818030Z [command]/usr/local/opt/ruby@2.7/bin/gem update fastlane -i /Users/runner/.gem-cache
2021-11-16T12:59:26.7363040Z Updating installed gems
2021-11-16T12:59:26.7464960Z Nothing to update
2021-11-16T12:59:26.7854080Z ##[warning]You specified releaseNotes - so you need to provide fastlane session, app specific password only won't work
2021-11-16T12:59:26.7890900Z [command]/usr/local/lib/ruby/gems/2.7.0/bin/fastlane pilot upload --api_key_path /Users/runner/work/_temp/api_keyMyKey.json -i /Users/runner/work/r1/a/$(Pipeline.Workspace)/ios_drop/myIpa.ipa --changelog MyChages
2021-11-16T12:59:26.7896820Z - App framework updated to version 2.x -a com.xxx.xxx --distribute_external true --groups AppGroups
2021-11-16T12:59:29.2059060Z [12:59:29]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
2021-11-16T12:59:34.0076090Z [12:59:34]: Sending anonymous analytics information
2021-11-16T12:59:34.0077420Z [12:59:34]: Learn more at https://docs.fastlane.tools/#metrics
2021-11-16T12:59:34.0079650Z [12:59:34]: No personal or sensitive data is sent.
2021-11-16T12:59:34.0080130Z [12:59:34]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
2021-11-16T12:59:34.0341370Z [12:59:34]: Creating authorization token for App Store Connect API
2021-11-16T12:59:39.6942440Z [12:59:39]: Ready to upload new build to TestFlight (App: 1************)...
2021-11-16T12:59:41.0207620Z [12:59:41]: Going to upload updated app to App Store Connect
2021-11-16T12:59:41.0211430Z [12:59:41]: This might take a few minutes. Please don't interrupt the script.
2021-11-16T13:00:45.2154010Z [13:00:45]: Transporter transfer failed.
2021-11-16T13:00:45.2255890Z [13:00:45]: 
2021-11-16T13:00:45.2357840Z [13:00:45]: **An error occurred while processing the json request. (1268)**
2021-11-16T13:00:45.3766610Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:44 UTC] <main> DBG-X:   parameter ErrorMessage = An error occurred while processing the json request. (1268)
2021-11-16T13:00:45.3814960Z 
2021-11-16T13:00:45.3817380Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:44 UTC] <main> DBG-X:   parameter ShouldUseRESTAPIs = false
2021-11-16T13:00:45.3822960Z 
2021-11-16T13:00:45.3825170Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:44 UTC] <main> DBG-X:   parameter Success = false
2021-11-16T13:00:45.3826310Z 
2021-11-16T13:00:45.3835020Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:44 UTC] <main> ERROR: An error occurred while processing the json request. (1268)
2021-11-16T13:00:45.3835420Z 
2021-11-16T13:00:45.3836250Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:44 UTC] <main> DBG-X: The error code is: 1268
2021-11-16T13:00:45.3836560Z 
2021-11-16T13:00:45.3837470Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:44 UTC] <main>  INFO: Done performing authentication.
2021-11-16T13:00:45.3837790Z 
2021-11-16T13:00:45.3838090Z [13:00:45]: [iTMSTransporter] 
2021-11-16T13:00:45.3838250Z 
2021-11-16T13:00:45.3838540Z [13:00:45]: [iTMSTransporter] 
2021-11-16T13:00:45.3838690Z 
2021-11-16T13:00:45.3838950Z [13:00:45]: [iTMSTransporter] 
2021-11-16T13:00:45.3839110Z 
2021-11-16T13:00:45.3839430Z [13:00:45]: [iTMSTransporter] Package Summary:
2021-11-16T13:00:45.3839630Z 
2021-11-16T13:00:45.3839890Z [13:00:45]: [iTMSTransporter]  
2021-11-16T13:00:45.3840050Z 
2021-11-16T13:00:45.3840460Z [13:00:45]: [iTMSTransporter] 1 package(s) were not uploaded because they had problems:
2021-11-16T13:00:45.3840750Z 
2021-11-16T13:00:45.3841890Z [13:00:45]: [iTMSTransporter] 	/var/folders/24/8k4gn/T/myIpa.ipa (/var/folders/24/8k4gn/T/DeveloperAPIUpload****/***.itmsp) - Error Messages:
2021-11-16T13:00:45.3842470Z 
2021-11-16T13:00:45.3842870Z [13:00:45]: [iTMSTransporter] 		An error occurred while processing the json request. (1268)
2021-11-16T13:00:45.3843140Z 
2021-11-16T13:00:45.3843900Z [13:00:45]: [iTMSTransporter] [2021-11-16 13:00:45 UTC] <main> DBG-X: Returning 1
2021-11-16T13:00:45.3844190Z 
2021-11-16T13:00:45.3844500Z [13:00:45]: iTunes Transporter output above ^
2021-11-16T13:00:45.3844930Z [13:00:45]: An error occurred while processing the json request. (1268)
2021-11-16T13:00:45.3845460Z Return status of iTunes Transporter was 1: An error occurred while processing the json request. (1268)
2021-11-16T13:00:45.3846420Z The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
2021-11-16T13:00:45.4166640Z 
2021-11-16T13:00:45.4167740Z [!] Error uploading ipa file: 
2021-11-16T13:00:45.4168310Z  
2021-11-16T13:00:45.4488750Z ##[error]Error: The process '/usr/local/lib/ruby/gems/2.7.0/bin/fastlane' failed with exit code 1

--- a/fastlane_core/lib/fastlane_core/itunes_transporter.rb
+++ b/fastlane_core/lib/fastlane_core/itunes_transporter.rb
@@ -150,7 +150,10 @@ module FastlaneCore
     end
 
     def file_upload_option(source)
-      if !File.directory?(source) && [".ipa", ".pkg", ".dmg", ".zip"].include?(File.extname(source).downcase)
+      ext = File.extname(source).downcase
+      is_asset_file_type = !File.directory?(source) && [".ipa", ".pkg", ".dmg", ".zip"].include?(ext)
+
+      if is_asset_file_type
         return "-assetFile #{source.shellescape}"
       else
         return "-f #{source.shellescape}"
@@ -492,7 +495,11 @@ module FastlaneCore
       # However, -assetFile requires -assetDescription if Linux or Windows
       # This will give the asset directly if macOS and asset_path exists
       # otherwise it will use the .itmsp package
-      actual_dir = if Helper.is_mac? && asset_path
+
+      force_itmsp = FastlaneCore::Env.truthy?("ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD")
+      can_use_asset_path = Helper.is_mac? && asset_path
+
+      actual_dir = if can_use_asset_path && !force_itmsp
                      # The asset gets deleted upon completion so copying to a temp directory
                      tmp_asset_path = File.join(Dir.tmpdir, File.basename(asset_path))
                      FileUtils.cp(asset_path, tmp_asset_path)

--- a/fastlane_core/lib/fastlane_core/itunes_transporter.rb
+++ b/fastlane_core/lib/fastlane_core/itunes_transporter.rb
@@ -150,10 +150,10 @@ module FastlaneCore
     end
 
     def file_upload_option(source)
-      if File.extname(source) == ".itmsp"
-        return "-f #{source.shellescape}"
-      else
+      if !File.directory?(source) && [".ipa", ".pkg", ".dmg", ".zip"].include?(File.extname(source).downcase)
         return "-assetFile #{source.shellescape}"
+      else
+        return "-f #{source.shellescape}"
       end
     end
 

--- a/fastlane_core/spec/itunes_transporter_spec.rb
+++ b/fastlane_core/spec/itunes_transporter_spec.rb
@@ -336,6 +336,27 @@ describe FastlaneCore do
               end
             end
           end
+
+          describe "with package_path and asset_path" do
+            describe "upload command generation" do
+              it 'generates a call to xcrun iTMSTransporter with -assetFile' do
+                expect(Dir).to receive(:tmpdir).and_return("/tmp")
+                expect(FileUtils).to receive(:cp)
+
+                transporter = FastlaneCore::ItunesTransporter.new(nil, nil, false, nil, jwt)
+                expect(transporter.upload(package_path: '/tmp/my.app.id.itmsp', asset_path: '/tmp/my_app.ipa')).to eq(java_upload_command(jwt: jwt, use_asset_path: true))
+              end
+            end
+
+            describe "upload command generation with ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD=true" do
+              it 'generates a call to xcrun iTMSTransporter with -assetFile' do
+                stub_const('ENV', { 'ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD' => 'true' })
+
+                transporter = FastlaneCore::ItunesTransporter.new(nil, nil, false, nil, jwt)
+                expect(transporter.upload(package_path: '/tmp/my.app.id.itmsp', asset_path: '/tmp/my_app.ipa')).to eq(java_upload_command(jwt: jwt, use_asset_path: false))
+              end
+            end
+          end
         end
       end
 

